pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -131,10 +131,8 @@ func (c *NetworkPolicyController) processClusterGroup(cg *crdv1beta1.ClusterGrou
 		for i := range cg.Spec.IPBlocks {
 			ipb, _ := toAntreaIPBlockForCRD(&cg.Spec.IPBlocks[i])
 			internalGroup.IPBlocks = append(internalGroup.IPBlocks, *ipb)
-			// CIDR format is already validated by the webhook
-			_, ipNet, _ := net.ParseCIDR(cg.Spec.IPBlocks[i].CIDR)
-			internalGroup.IPNets = append(internalGroup.IPNets, *ipNet)
 		}
+		internalGroup.IPNets = computeEffectiveIPNetForIPBlocks(cg.Spec.IPBlocks)
```

Comment:
so if I'm understanding correctly, the "effective" CIDR list is computed by the controller for Groups / ClusterGroups, and by the agent for policy peers (K8s NP or Antrea-native NP)? does that discrepancy make sense?

Reply:
The "effective" CIDR list is computed by the controller for Groups / ClusterGroup _only_ for membership and ip association query purposes. When groups are referred to in an Antrea-native policy, the CIDR list is still computed at agent level, unless we want to make a bigger controlplane API change so that the information gets carried over to agent

Reply:
In controlplane/types.go:
```
type NetworkPolicyPeer struct {
	AddressGroups []string
	IPBlocks []IPBlock
	FQDNs []string
	ToServices []ServiceReference
	LabelIdentities []uint32
}

type IPBlock struct {
	CIDR IPNet
	Except []IPNet
}
```
So in order to unify the computation at the controller side, this NetworkPolicyPeer types needs to be changed to pass IPNet instead of IPBlocks, which I think will be an overkill for this feature's purpose

Reply:
I didn't realize that the `IPNets` field was only used locally in the controller (for membership queries) and not transmitted to the agent. You can disregard my comment in that case.